packages/store/src/utils.js

```diff
@@ -1,8 +1,116 @@
 // @ts-check
+import { E } from '@endo/eventual-send';
+import { isPromise } from '@endo/promise-kit';
 
-const { getPrototypeOf, create, fromEntries } = Object;
+const { getPrototypeOf, create, entries, fromEntries } = Object;
 const { ownKeys, apply } = Reflect;
 
+const { details: X } = assert;
+
+/**
+ * By analogy with how `Array.prototype.map` will map the elements of
+ * an array to transformed elements of an array of the same shape,
+ * `objectMap` will do likewise for the string-named own enumerable
+ * properties of an object.
+ *
+ * Typical usage applies `objectMap` to a CopyRecord, i.e.,
+ * an object for which `passStyleOf(original) === 'copyRecord'`. For these,
+ * none of the following edge cases arise. The result will be a CopyRecord
+ * with exactly the same property names, whose values are the mapped form of
+ * the original's values.
+ *
+ * When the original is not a CopyRecord, some edge cases to be aware of
+ *    * No matter how mutable the original object, the returned object is
+ *      hardened.
+ *    * Only the string-named enumerable own properties of the original
+ *      are mapped. All other properties are ignored.
+ *    * If any of the original properties were accessors, `Object.entries`
+ *      will cause its `getter` to be called and will use the resulting
+ *      value.
+ *    * No matter whether the original property was an accessor, writable,
+ *      or configurable, all the properties of the returned object will be
+ *      non-writable, non-configurable, data properties.
+ *    * No matter what the original object may have inherited from, and
+ *      no matter whether it was a special kind of object such as an array,
+ *      the returned object will always be a plain object inheriting directly
+ *      from `Object.prototype` and whose state is only these new mapped
+ *      own properties.
+ *
+ * With these differences, even if the original object was not a CopyRecord,
+ * if all the mapped values are Passable, then the returned object will be
+ * a CopyRecord.
+ *
+ * @template {string} K
+ * @template T
+ * @template U
+ * @param {Record<K,T>} original
+ * @param {(value: T, key?: string) => U} mapFn
```

Comment:
minor
```suggestion
 * @param {(value: T, key?: K) => U} mapFn
```

Reply:
When I do this, I get
```
src/utils.js:60:53 - error TS2345: Argument of type 'string' is not assignable to parameter of type 'K'.
  'string' is assignable to the constraint of type 'K', but 'K' could be instantiated with a different subtype of constraint 'string'.

60   const mapEnts = ents.map(([k, v]) => [k, mapFn(v, k)]);
                                                       ~
```
(this is probably how I ended writing the less natural type, but frankly I don't remember)

In any case, what do you recommend I do about this complaint?

Reply:
After double checking that `entries` (used inside `objectMap`) enumerates only string-named properties, I just removed the `K` type parameter, replacing it everywhere with `string`. PTAL.

Reply:
Since 
   * you've already approved
   * my removal of `K` seems right to both me and typescript
   * I can now get this PR on the merge train
   * the only possible controversy is static rather than dynamic behavior

I'm labeling it with automerge:squash until I hear otherwise. If this merges before you suggest a different fix, we can always to that one after merge

Reply:
The way now loses the keys. E.g. `objectMap({a: 1}, v=>v).whatever` won't error.

I suggest [this](https://www.typescriptlang.org/play?target=99&filetype=js#code/PQKhCgAIUgBAXApgWwA4BsCGTIG8D2ARgFaIDG8AvpACpQyyqYBOmyeN1+zAlgOY8AdpnT04SNFhwAlSMkypIzRAGcArunhjGLNngAUANxFrEALloAaSAGtEATwD8Fu-fwAzWgEpIAXgB8kNLU8qgAYoLayvBqzIIqeLiQANoACpBCtg4etAC6FsGU9MDgiAAeqNzwkGT48dVEpBQAsgp+kPrc-EIi1qERPgF4UDV1KtWIgvAJvpAA8iTk8AB0k-C8qp28AsLoXgDcI7X1cgoAolMzkGsqy6H6+sk21oa5g4FPfQoRRtY2Xm9DpAlIgYnF5osKMt3Mx8MgLuseJtQgiVAdwJRDuBjuNII12nhMBYAIzWQgWABMgkx2LG1QAJvg1IR0IgFsR2o0lq1UJ0SC8RH5AsY9ljgMBICoABZM9D067MWHMSCENQM-CqQQAcmqAHcePApVl7BZxrxBHxwIzmaz2XceCoVEI+IdrSy2SRlkA) instead.

<img width="542" alt="Screen Shot 2022-08-18 at 4 13 15 PM" src="https://user-images.githubusercontent.com/21505/185510808-f60f4f49-ad59-471d-a43a-7973cb104ac8.png">